In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import deep_nn as nn

Let's analyze the data a little bit...

First of all I'd like to give a small preview of the test data, let's print some basic info.

In [5]:
data = pd.read_csv("train.csv")

print("SHAPE", data.shape)
print(data.head())

SHAPE (42000, 785)
   label  pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  \
0      1       0       0       0       0       0       0       0       0   
1      0       0       0       0       0       0       0       0       0   
2      1       0       0       0       0       0       0       0       0   
3      4       0       0       0       0       0       0       0       0   
4      0       0       0       0       0       0       0       0       0   

   pixel8    ...     pixel774  pixel775  pixel776  pixel777  pixel778  \
0       0    ...            0         0         0         0         0   
1       0    ...            0         0         0         0         0   
2       0    ...            0         0         0         0         0   
3       0    ...            0         0         0         0         0   
4       0    ...            0         0         0         0         0   

   pixel779  pixel780  pixel781  pixel782  pixel783  
0         0         0         0

Ok, so it seems we have 42000 training samples, and every trainign sample consist on 785 values. 

The first of them is the label (0..9) and the rest are the pixel values of the image (0..255)

Let's analyze the labels a little bit...

In [18]:
labels = pd.read_csv("train.csv").as_matrix()[:, :1].reshape(-1)

print(labels[:20])

[1 0 1 4 0 0 7 3 5 3 8 9 1 3 3 1 2 0 7 5]


Let's count how many samples of each number do we have...

In [19]:
for i in range(10):
    count = np.count_nonzero(labels == i)
    print("The number "+str(i)+" appears "+str(count)+" times")

The number 0 appears 4132 times
The number 1 appears 4684 times
The number 2 appears 4177 times
The number 3 appears 4351 times
The number 4 appears 4072 times
The number 5 appears 3795 times
The number 6 appears 4137 times
The number 7 appears 4401 times
The number 8 appears 4063 times
The number 9 appears 4188 times


It seems like the distribution is pretty homogeneous. 1 is the most frequent label, while 5 is the least frequent, but that should not be an issue...